app/src/main/java/com/voltix/wallet/data/on_board/db/entity/VaultEntity.kt

```diff
@@ -0,0 +1,120 @@
+package com.voltix.wallet.data.on_board.db.entity
+
+import androidx.room.ColumnInfo
+import androidx.room.Embedded
+import androidx.room.Entity
+import androidx.room.ForeignKey
+import androidx.room.Index
+import androidx.room.PrimaryKey
+import androidx.room.Relation
+import com.voltix.wallet.models.Chain
+import java.util.Date
+
+@Entity(
+    tableName = "Vault",
+    indices = [Index(value = ["local_party_id"], unique = true)],
+    foreignKeys = [
+        ForeignKey(
+            entity = KeyShareEntity::class,
+            parentColumns = ["vault_local_party_id"],
+            childColumns = ["local_party_id"],
+            onDelete = ForeignKey.CASCADE,
+            onUpdate = ForeignKey.CASCADE
+        ),
+        ForeignKey(
+            entity = CoinEntity::class,
+            parentColumns = ["vault_local_party_id"],
+            childColumns = ["local_party_id"],
+            onDelete = ForeignKey.CASCADE,
+            onUpdate = ForeignKey.CASCADE
+        )
+    ]
+)
+
+data class VaultEntity(
+    @PrimaryKey()
+    @ColumnInfo("local_party_id")
```

Comment:
local party id will not be the primary key , since one device can have multiple vaults. 

PubKeyECDSA should be unique